projectroles/app_settings.py

```diff
@@ -260,7 +285,24 @@ def get_default(cls, app_name, setting_name, post_safe=False):
             app_settings = app_plugin.app_settings
 
         if setting_name in app_settings:
-            if app_settings[setting_name]['type'] == 'JSON':
+            if callable(app_settings[setting_name].get('default')):
+                try:
+                    callable_setting = app_settings[setting_name]['default']
+                    if app_settings[setting_name].get('options'):
```

Comment:
There seems to be a mistake here. The callable default value and callable options should be handled separaterly. Here we seem to get the default value using a callable if `options` are set in the app settings definition. These should not be related. Or am I misundestanding something?

Here's how it's supposed to work:

- If `default` is a callable, call the default callable to get the default value
- If `options` is a callable, call the options callable to get the options.

There should be no direct relationship between the two. Both, either or neither of them could be a callable for a specific app setting.

Reply:
The thing here is, that in exmple setting I have following:
```
'project_callable_setting_options': {
            'scope': SODAR_CONSTANTS['APP_SETTING_SCOPE_PROJECT'],
            'type': 'STRING',
            'label': 'Callable setting with options',
            'default': get_example_setting_options,
            'options': get_example_setting_options,
            'description': 'Example callable project setting with options',
            'user_modifiable': True,
        }
```
In that case, when I am passing default value for setting with options, I need to pass option functios, that will return a default value. There was before `get_example_setting_default` function, but as you said it has to be independant with option function.